cli/types/cypress.d.ts

```diff
@@ -2,6 +2,7 @@
 /// <reference path="./cypress-npm-api.d.ts" />
 /// <reference path="./cypress-eventemitter.d.ts" />
 /// <reference path="./cypress-type-helpers.d.ts" />
+// Minimum TypeScript Version: 4.1
```

Comment:
If this places a minimum typescript requirement in our shipped types, what happens if someone is on an older version? Will this cause issues? Wondering how passive this is.

Reply:
Trying to figure this out still... I wonder if we can bump our minimum TS version? 5.x is out now -- 3.4 came out in Mar 2019.

Reply:
FWIW, the [Definitely Typed support window](https://github.com/DefinitelyTyped/DefinitelyTyped#support-window) is 4.3 - 5.0 (they support each version for 2 years from its release.

Reply:
Oh ok, that might be the issue.

I think we might want to look into bumping out minimal TS version as well. We will pick this up in our next sprint of work (2 week blocks, starting new one tomorrow). So this won't be in the next release (tomorrow) but the following one (2 weeks time).

Reply:
I wonder if we can go to 4.x, considering 5.x is now out.

I'm not entirely clear on what makes this a breaking change - any context?

Reply:
Posted in the issue: https://github.com/cypress-io/cypress/issues/24875#issuecomment-1507662491